opencart 2.2.0.0/upload/admin/model/smailyforopencart/admin.php

```diff
@@ -16,6 +16,10 @@ public function install() {
             PRIMARY KEY (customer_id)
             )"
         );
+        // Save 1970-1-1 sync time to settings.
+        $this->load->model('setting/setting');
+        $data['module_smaily_for_opencart_sync_time'] = date('Y-m-d H:i:s', 0);
+        $this->model_setting_setting->editSetting('module_smaily_for_opencart_sync', $data);
```

Comment:
You don't need to have `1970-01-01` stored in the settings. `getSetting` returns empty array when setting key does not exist, or if you are using `getSettingValue` a `NULL` is returned. Both can be handled easily when getting the last synchronization time and/or fetching customers.